### PR TITLE
feat(behavior_path_palnner): update geometric parallel parking for pull_out module

### PR DESCRIPTION
**Pull request by kosuke55**
Signed-off-by: kosuke55 <kosuke.tnp@gmail.com>

## Description

This PR is the part of https://github.com/autowarefoundation/autoware.universe/pull/1438. 
Update geoemetric_parallel_parking for pull_out and update pull_over accordingly.
The behavior of pull_over will not change.
- add `planDeparting` which is the reverse of parking.
- make some variable parameters.
- refactoring
- change pull over max_steer_rad 40deg to 20deg(Some individuals are unable to steer 40deg when stopped.)
- fix lane_id of arc path. add road and shoulder lane ids to arc path.

## Related links

part of https://github.com/autowarefoundation/autoware.universe/pull/1438
merge https://github.com/autowarefoundation/autoware.universe/pull/1571 first : **done**
launcher: https://github.com/tier4/autoware_launch/pull/431

## Tests performed

psim

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
@kosuke55 can you update readme?

**Comment by kosuke55** [comment]
@yukkysaito 
yes, I updated README in 
https://github.com/autowarefoundation/autoware.universe/pull/1578 for pull_over
https://github.com/autowarefoundation/autoware.universe/pull/1587 for pull_out

File: planning/behavior_path_planner/src/scene_module/utils/geometric_parallel_parking.cpp [hunk 1]
```diff
@@ -68,83 +80,265 @@ PathWithLaneId GeometricParallelParking::getCurrentPath() const
...
+        path.points.at(i).point.longitudinal_velocity_mps = 0.0;
+      } else {
+        path.points.at(i).point.longitudinal_velocity_mps = velocity;
+      }
+    }
+  }
+}
+
+std::vector<PathWithLaneId> GeometricParallelParking::generateParkingPaths(
+  const Pose & start_pose, const Pose & goal_pose, const double R_E_r,
```
Comment (yukkysaito) [inline planning/behavior_path_planner/src/scene_module/utils/geometric_parallel_parking.cpp:142]: what mean R_E_r?

Comment (kosuke55) [reply planning/behavior_path_planner/src/scene_module/utils/geometric_parallel_parking.cpp:142]: This comes from the referenced paper. This is the turning radius of the right turn of the base link.

https://github.com/tier4/autoware.universe/blob/docs/update_pull_over_docs/planning/behavior_path_planner/README.md#references--external-links

image

Comment (yukkysaito) [reply planning/behavior_path_planner/src/scene_module/utils/geometric_parallel_parking.cpp:142]: I see. can you add the reference to README?

Comment (kosuke55) [reply planning/behavior_path_planner/src/scene_module/utils/geometric_parallel_parking.cpp:142]: I have added it to the README of behavior_path as linked above.

Comment (yukkysaito) [reply planning/behavior_path_planner/src/scene_module/utils/geometric_parallel_parking.cpp:142]: :+1:

**Review by yukkysaito — APPROVED** [review]
LGTM
pls get approval by @TakaHoribe

**Comment by kosuke55** [comment]
@yukkysaito @TakaHoribe 
Thanks, please check also launcher https://github.com/tier4/autoware_launch/pull/431

**Review by TakaHoribe — APPROVED** [review]
LGTM. Memo: this PR enables the geometryc_parking in the pull_out module as well as the pull_over.

**Pull request merged**